### ISSUE TITLE
[feature request] Lua module emulator

### ISSUE THREAD

**Issue by alzamer2** [dazedcat19/FMD2#520]
when i tried to wrote lua module for site: i encounter these problem:
1- didn't know the lua module loading form
2- didn't know function and variable that can be written in lua file and output that program except to get
3- needed to restart FMD2 multi times to check and test lua file

so why not create other interface that has core function of FMD2 with striping other stuff, and it can reload lua file with out closing and has log that test and show errors directly?

here example of what it will look like
![image](https://user-images.githubusercontent.com/4492974/116512605-fc085300-a8d0-11eb-865a-27f507315cf9.png)

notes:
1-tests that need not found/error function are disabled
2- its better if emulator tools function had some limit that will stop tests automatically as some sites has huge database like mangadex
3- it will be plus if you add help that list the function:
    a-description of function.
    b-what variable are available to use.
    c-expected output(type like str/bin/dict..etc )

ps: i know that "test getting titles from site" and "test load series" should be disable but i shoed it like that as example

**Comment by Slasar41** [comment]
FMD2 could read the changes made to the lua files without needing to reload the app, see the first bullet of README.
But I've tried it and it's not working. Could be a bug or something is wrong on my end.

**Comment by alzamer2** [comment]
you mean --lua-dofile?
from the description i thought it was so it will ignore lua update from github and focus on lua files in harddisk

i will try it next time(hope it will be fixed by then) when i write lua file

side not in python there is an external module called "[browser_cookie3](https://github.com/borisbabic/browser_cookie3)" which can used to grab cookies for login and cloudflare by-pass from installed browser (firefox,chrome..etc), can this module used as ref or imported to code?

**Comment by Slasar41** [comment]
> you mean --lua-dofile?
> from the description i thought it was so it will ignore lua update from github and focus on lua files in harddisk

Yes, you can read the conversation with the dev [here](https://github.com/dazedcat19/FMD2/issues/59).
> side not in python there is an external module called "[browser_cookie3](https://github.com/borisbabic/browser_cookie3)" which can used to grab cookies for login and cloudflare by-pass from installed browser (firefox,chrome..etc), can this module used as ref or imported to code?

Is it different with selenium? https://github.com/dazedcat19/FMD2/issues/120
FMD2 already support it, although the browser get stuck on cloudflare page.
![image](https://user-images.githubusercontent.com/28888646/116772282-103a8480-aa78-11eb-8232-69e52c0da097.png)

Also, FMD2 immediately get blocked by cloudflare (error 1020) on websites like komikcast, reaperscans, batoto, tumangaonline, etc
We need to figure it out what FMD2 is lacking so it'll able to be recognized by cloudflare again.

**Comment by alzamer2** [comment]
Cookies_brower3 is not same as selenium

selenium use browser as interface to open web page, while Cookies_brower3 open database of browser and get cookies from it ( no need to open browser)

As for for error 1020 I am not sure but I had similar problem when I used cookies from Firefox but used user agent for chrome, so maybe this error happen because FMD2 user agent not same as the user agent of client that selenium use

**Comment by Slasar41** [comment]
> Cookies_brower3 is not same as selenium
> 
> selenium use browser as interface to open web page, while Cookies_brower3 open database of browser and get cookies from it ( no need to open browser)

I see, if you want to fix cloudflare issue we can talk it in the discord server.

> As for for error 1020 I am not sure but I had similar problem when I used cookies from Firefox but used user agent for chrome, so maybe this error happen because FMD2 user agent not same as the user agent of client that selenium use

Selenium is not involved, it's just FMD2 trying to access the websites but got blocked by error 1020.
I guess FMD2 missing some features like maybe the headers or something and failed to mimic the real browser that required by cloudflare recently. Because a few weeks ago those websites still works fine in FMD2.

And regarding your lua emulator app, maybe you can publish it to your github and other people can use it too.
